From 2f845fea72bd128ec5223bc1b34b114da8f4d40b Mon Sep 17 00:00:00 2001
From: Shadi Ammouri <shadi@marvell.com>
Date: Mon, 9 Oct 2017 10:21:27 +0300
Subject: [PATCH 2020/2241] uio: Add property to control generic uio device
 name

Add a new property "uio-name" to contrl the name of the uio device
created by the generic uio driver.

Change-Id: I3ad22f396d2aebfa5a2a61e6741f225f1f1a7528
Signed-off-by: Shadi Ammouri <shadi@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/45069
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
---
 .../devicetree/bindings/uio/uio_pdrv_genirq.txt       | 19 +++++++++++++++++++
 drivers/uio/uio_pdrv_genirq.c                         |  5 ++++-
 2 files changed, 23 insertions(+), 1 deletion(-)
 create mode 100644 Documentation/devicetree/bindings/uio/uio_pdrv_genirq.txt

diff --git a/Documentation/devicetree/bindings/uio/uio_pdrv_genirq.txt b/Documentation/devicetree/bindings/uio/uio_pdrv_genirq.txt
new file mode 100644
index 0000000..e63e409
--- /dev/null
+++ b/Documentation/devicetree/bindings/uio/uio_pdrv_genirq.txt
@@ -0,0 +1,19 @@
+This file describes the changes done to the uio_pdrv_genirq driver, along with
+the new device-tree properties required to control these features.
+
+Added properties:
+
+- uio-name: Holds the name of the created uio device.
+In case this property exists, then it will be used as the uio device name
+(/sys/class/uio/uioN/name), in case it's not defined, the driver will use
+the node's name as the uio device name.
+
+Example:
+	dma_xor@6a0000 {
+        	compatible = "marvell,mv-xor-v2";
+		reg = <0x6a0000 0x1000>,
+	              <0x6b0000 0x1000>;
+		musdk-mode;
+        	uio-name = "uio_dma_xor_4";
+	        status = "okay";
+	};
diff --git a/drivers/uio/uio_pdrv_genirq.c b/drivers/uio/uio_pdrv_genirq.c
index f598ecd..12c90a6 100644
--- a/drivers/uio/uio_pdrv_genirq.c
+++ b/drivers/uio/uio_pdrv_genirq.c
@@ -118,7 +118,10 @@ static int uio_pdrv_genirq_probe(struct platform_device *pdev)
 			dev_err(&pdev->dev, "unable to kmalloc\n");
 			return -ENOMEM;
 		}
-		uioinfo->name = pdev->dev.of_node->name;
+
+		if (of_property_read_string(pdev->dev.of_node, "uio-name",
+					&uioinfo->name) != 0)
+			uioinfo->name = pdev->dev.of_node->name;
 		uioinfo->version = "devicetree";
 		/* Multiple IRQs are not supported */
 	}
-- 
2.7.4

